### PR TITLE
[WIP] Inert modal overlay

### PR DESCRIPTION
**Pull request by c0g1t8**
# Pull Request

For #782. 

## 📖 Description

Adds the option to prevent dismissing a modal dialog by clicking or touching the overlay.

It is implemented as an additional option and is a non-breaking change.

### 🎫 Issues

#782

## 👩‍💻 Reviewer Notes

Option name - **InertModalOverlay** is there a better name.

### General

- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [X] I have modified an existing component
- [ ] I have validate [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

**Comment by c0g1t8** [comment]
@vnbaaij 

Thoughts and comments are welcome.

**Comment by dvoituron** [comment]
You could use the **Modal** attribute and the **FluentDialogHeader** with ´ShowDismiss="false"´

**Comment by vnbaaij** [comment]
Setting `ShowDismiss="false"` does not prevent the dialog from actually being dismissed by clicking on the overlay.

As it is mostly about dismissing the dialog (scrolling the background can be turned on/off since 3.1's `PreventScroll` addition) Why not just call the new parameter `PreventDismiss` (default = "false)?

**Comment by dvoituron** [comment]
> Setting `ShowDismiss="false"` does not prevent the dialog from actually being dismissed by clicking on the overlay.
> 
> As it is mostly about dismissing the dialog (scrolling the background can be turned on/off since 3.1's `PreventScroll` addition) Why not just call the new parameter `PreventDismiss` (default = "false)?

By using this **header** and **footer** in your dialog file, the user **cannot** close the dialog by clicking outside the dialog or on a button (removed with this code). The only option (probably to be corrected in another PR) is to press the `ESC` key.

```xml
<FluentDialogHeader ShowDismiss="false" />
<FluentDialogFooter Visible="false" />
```

File: src/Core/Components/Dialog/Parameters/DialogParameters.cs [hunk 1]
```diff
@@ -20,10 +20,16 @@ public class DialogParameters : ComponentParameters, IDialogParameters
 
     /// <summary>
     /// Determines if the dialog is modal. Defaults to true.
-    /// When true, clicking outside the dialog will dismiss the dialog.
+    /// Partially obscures the area around the dialog.
```
Comment (vnbaaij) [inline src/Core/Components/Dialog/Parameters/DialogParameters.cs:23]: Remove 'Partially' please

File: src/Core/Components/Dialog/Parameters/DialogParameters.cs [hunk 2]
```diff
@@ -20,10 +20,16 @@ public class DialogParameters : ComponentParameters, IDialogParameters
...
     /// </summary>
     public bool? Modal { get; set; } = true;
 
+    /// <summary>
+    /// Determines if a modal dialog is dismissible by clicking
+    /// outside the dialog. Defaults to false.
+    /// </summary>
+    public bool? InertModalOverlay { get; set; } = false;
```
Comment (vnbaaij) [inline src/Core/Components/Dialog/Parameters/DialogParameters.cs:31]: Rename to `PreventDismissOnOverlayClick` please to make it more clear what it is and does.

**Review by vnbaaij — CHANGES_REQUESTED** [review]

**Comment by vnbaaij** [comment]
Hi Gary,

Denis and I discussed a bit more on this internally and we think this is a good thing to add. Can you change the parameter name to `PreventDismissOnOverlayClick`. We feel that makes it's use a bit more clear. I started a PR review for this

**Comment by StevenRasmussen** [comment]
This is very timely ;)  I was just looking for this functionality. Thanks!

**Comment by vnbaaij** [comment]
I'm pulling this one in and make the requested changes myself. Want to get 3.1.1 out asap

**Pull request merged**